src/operator/windowTime.ts

```diff
@@ -83,17 +127,35 @@ interface CreationState<T> {
   scheduler: Scheduler;
 }
 
+class CountedSubject<T> extends Subject<T> {
+  private numberOfNextedValues: number;
+  constructor() {
+    super();
+    this.numberOfNextedValues = 0;
+  }
+
+  next(value?: T) {
+    super.next(value);
+    this.numberOfNextedValues++;
```

Comment:
This increment needs to happen before we `super.next(value)` otherwise someone can recursively reenter and never reach the max count because it hasn't been incremented yet.